### PR TITLE
test(devices): increased code coverage for devices

### PR DESCRIPTION
**Pull request by Mahendra-intel**
## PR Checklist

- [ ] Unit Tests have been added for new changes
- [ ] API tests have been updated if applicable
- [ ] All commented code has been removed
- [ ] If you've added a dependency, you've ensured license is compatible with Apache 2.0 and clearly outlined the added dependency.

### CODE REVIEW

**Review by rsdmike — APPROVED** [review]

**Review by rsdmike — CHANGES_REQUESTED** [review]
Almost there, update tests to ensure the correct route/API is being called. Be sure and use a variable for the repetitive part of the URL

File: src/app/devices/device-toolbar/device-toolbar.component.spec.ts [hunk 1]
```diff
@@ -6,12 +6,14 @@ import { DevicesService } from '../devices.service'
 import { DeviceToolbarComponent } from './device-toolbar.component'
 import { ActivatedRoute } from '@angular/router'
 import { of } from 'rxjs'
+import { MatDialog } from '@angular/material/dialog'
+
 describe('DeviceToolbarComponent', () => {
```
Comment (rsdmike) [inline src/app/devices/device-toolbar/device-toolbar.component.spec.ts:11]: Still a bit more to go for branches on Device Toolbar:
image

File: src/app/devices/devices.service.spec.ts [hunk 2]
```diff
@@ -4,24 +4,407 @@
 **********************************************************************/
 import { TestBed } from '@angular/core/testing'
 import { Router } from '@angular/router'
+import { of, throwError } from 'rxjs'
+import { environment } from 'src/environments/environment'
 import { AuthService } from '../auth.service'
 
 import { DevicesService } from './devices.service'
 
 describe('DevicesService', () => {
```
Comment (rsdmike) [inline src/app/devices/devices.service.spec.ts:13]: getHardwareINformation tests are missing

url condition for actions >= 100 is missing

File: src/app/devices/devices.component.spec.ts [hunk 3]
```diff
@@ -18,9 +18,10 @@ describe('DevicesComponent', () => {
   let getDevicesSpy: jasmine.Spy
   let getTagsSpy: jasmine.Spy
   let getPowerStateSpy: jasmine.Spy
+  let sendPowerActionSpy: jasmine.Spy
 
   beforeEach(async () => {
```
Comment (rsdmike) [inline src/app/devices/devices.component.spec.ts:23]: tagChange, masterToggle, resetResponse, bulkPowerAction tests aren't tested either at all or not completely

image

**Review by rsdmike — CHANGES_REQUESTED** [review]
still a bit more to go on Devices, good work so far! Be sure and view the coverage html file to determine how much you've covered vs how much you're missing.

**Comment by rsdmike** [comment]
what about device-user-consent?

File: src/app/devices/devices.service.spec.ts (hunk 2 above)
Comment (Mahendra-intel) [reply src/app/devices/devices.service.spec.ts:13]: add test cases for  getHardwareinformation and actions >= 100

File: src/app/devices/devices.component.spec.ts (hunk 3 above)
Comment (Mahendra-intel) [reply src/app/devices/devices.component.spec.ts:23]: resolved

File: src/app/devices/devices.service.spec.ts (hunk 2 above)
Comment (Mahendra-intel) [reply src/app/devices/devices.service.spec.ts:13]: done

File: src/app/devices/device-toolbar/device-toolbar.component.spec.ts (hunk 1 above)
Comment (Mahendra-intel) [reply src/app/devices/device-toolbar/device-toolbar.component.spec.ts:11]: done

**Review by rsdmike — APPROVED** [review]

**Pull request merged**